packages/core/src/com/communication.ts

```diff
@@ -340,46 +347,53 @@ export class Communication {
                 this.handleReady(message);
                 break;
             case 'dispose':
-                if (message.from !== this.rootEnvId) {
-                    this.clearEnvironment(message.origin, message.from);
-                }
+                this.handleDispose(message);
                 break;
             case 'status':
-                this.handleStatus(message);
-                break;
-            default:
+                this.handleStatusRequest(message);
                 break;
+            default: {
+                const _exhaustiveCheck: never = message;
+            }
         }
     }
 
+    /**
+     * debug method: get all environments statuses
+     */
     public async getAllEnvironmentsStatus() {
-        const pending: Promise<ReturnType<Communication['getComStatus']>>[] = [];
-        const localStats = this.getComStatus();
-        // not us
-        const envs = Object.keys(this.environments).filter((envId) => envId !== '*' && envId !== this.rootEnvId);
-        for (const envId of envs) {
-            const { promise, resolve, reject } = deferred<ReturnType<Communication['getComStatus']>>();
-            const callbackId = this.idsCounter.next(this.messageIdPrefix);
-            this.callWithCallback(
+        type Status = ReturnType<typeof this.getStatus>;
+        const statuses: Promise<Status>[] = [];
+        const environmentsExcludingOwn = Object.keys(this.environments).filter(
+            (envId) => envId !== '*' && envId !== this.id,
+        );
+
+        for (const envId of environmentsExcludingOwn) {
+            const { promise, resolve, reject } = deferred<Status>();
+            const callbackId = this.idsCounter.next(this.callbackIdPrefix);
+            this.callWithCallback({
                 envId,
-                {
-                    from: this.rootEnvId,
-                    to: envId,
+                message: {
                     type: 'status',
-                    origin: this.rootEnvId,
+                    origin: this.id,
+                    from: this.id,
+                    to: envId,
                     callbackId,
                 },
                 callbackId,
                 resolve,
                 reject,
-            );
-            pending.push(promise);
+            });
+            statuses.push(promise);
         }
-        return Promise.allSettled(pending).then((results) => {
-            const statuses: Record<string, ReturnType<Communication['getComStatus']> | PromiseRejectedResult> = {};
+
+        return Promise.allSettled(statuses).then((results) => {
+            const statuses: Record<string, Status | PromiseRejectedResult> = {
+                [this.id]: this.getStatus(),
```

Comment:
This moves the time of the snapshot to after we take all other statuses. I'm not sure that it's matters. but does it?

Reply:
I wouldn't think it would... get status would be invoked on line 392 when creating an object on 391. before it did happen on 357, but between that and now only things that are happening are registering status calls to others; not waiting to get all those statuses. 

But it is indeed change in how things work, so I will revert to how it was. Great catch thank you!